### ISSUE TITLE
Do not rebuild native extensions on every install for git dependencies

### ISSUE THREAD

**Issue by grosser** [bundler/bundler#3014]
given a Gemfile with a native extensions
```
source "https://rubygems.org"
gem "rinku",                :git => "https://github.com/shajith/rinku.git", :tag => "v1.7.2-unicode"
```

the native extension should not be rebuild if it already was build the last time around

```
bundle --local --path vendor/bundle --quiet
```

basically make lib/bundler/source/path.rb:188 smarter

this should bring extra speed when bundling + allow to re-bundle without having a running app try to load a invalid/missing c extension

**Comment by egonSchiele** [comment]
+1, this is an issue for us too.

**Comment by moeffju** [comment]
:+1: We run into this or a similar issue with the `debase` gem. When using it from a git repository and running `bundle install`, the following artifacts get dropped in our `vendor/cache`:

```
?? vendor/cache/debase-264cddf71b48/ext/.RUBYARCHDIR.time
?? vendor/cache/debase-264cddf71b48/lib/debase_internals.bundle
```

We have `.gitignore`d the `vendor/cache/extensions/` directory since it is architecture- and sometimes machine-specific, but debase does not put its C extension there.

**Comment by Intrepidd** [comment]
:+1: bon courage working on this !

**Comment by segiddins** [comment]
@indirect / @andremedeiros this can be closed as of 1.12.0.pre.2, no?

**Comment by indirect** [comment]
I think so, yes.
